sourcegraph/sourcegraph
# [Svelte] Add refs picker to the repository page

Closes https://github.com/sourcegraph/sourcegraph/issues/61622

Note that designs are not final, this is just a version that I could come up with a bit faster since most of the UI components haven't been implemented by new specifications from Figma. (This should be good enough for the milestone 2 checkpoint but let me know if something important is missing)

| Modes | Image |
| ---- | ---- |
| Branch | <img  width="400" alt="Screenshot 2024-04-12 at 15 25 38" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/6b65bc34-02d4-4a8b-992c-4c90c8b48efd"> |
| Tag | <img   width="400" alt="Screenshot 2024-04-12 at 15 25 50" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/50f90dc0-fdd9-4915-8cbb-cc03702d9efd"> |
| Commit | <img   width="400" alt="Screenshot 2024-04-12 at 15 26 04" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/cbe0ad6f-d808-49ab-b6c3-4872af848f8b"> |

## Todo 
- [x] Add basic picker functionality 
- [x] Add combobox-like navigation to the picker suggestions
- [x] Support dark theme
- [x] Fix problems with melt-ui and our own popover functionality overlap 
- [ ] Add keyboard shortcuts for switching between tabs
- [ ] Add highlight words matching logic (most likely will ship it as separate PR) 

## Test plan
- Go to the blob or repository page
- Check branch, tag, and commit pickers
